Mage.Tests/src/test/java/org/mage/test/cards/single/dom/SimpleDominariaCards.java

```diff
@@ -70,21 +70,16 @@ public void testKnightOfGraceBlackAbility(){
         addCard(Zone.BATTLEFIELD, playerA, "Knight of Grace");
         addCard(Zone.BATTLEFIELD, playerB, "Avatar of Woe");
 
+        setStrictChooseMode(true);
+
         activateAbility(1, PhaseStep.PRECOMBAT_MAIN, playerB, "{T}");
-        addTarget(playerB, "Knight of Grace");
         setStopAt(1, PhaseStep.POSTCOMBAT_MAIN);
 
-        try {
-            execute();
-            Assert.fail("must throw exception on execute");
-        } catch (Throwable e) {
-            if (!e.getMessage().contains("setup good targets")) {
-                Assert.fail("must throw error about bad targets, but got:\n" + e.getMessage());
-            }
-        }
+        execute();
+        assertAllCommandsUsed();
 
-        assertTapped("Avatar of Woe", false);
-        assertGraveyardCount(playerA, "Knight of Grace", 0);
+        assertGraveyardCount(playerA, "Knight of Grace", 0); // Hexproof from black
+        assertGraveyardCount(playerB, "Avatar of Woe", 1);   // Only target and autochosen, autokills itself.
```

Comment:
> autokills itself

If a real user clicks on the card and kills itself without warning -- it's a really bad UX thing. See comment about UX options.

Reply:
I agree. I'll start a list above.

Reply:
Is there a good example of the server reading in user settings and acting on them server-side for something like this?